noir-projects/noir-contracts/contracts/account/schnorr_account_contract/src/main.nr

```diff
@@ -144,10 +143,10 @@ pub contract SchnorrAccount {
         // Compute the nullifier and check if it is spent
         // This will BLINDLY TRUST the oracle, but the oracle is us, and
         // it is not as part of execution of the contract, so we are good.
-        let nullifier = compute_authwit_nullifier(context.this_address(), inner_hash);
+        let nullifier = compute_authwit_nullifier(self.address, inner_hash);
         let siloed_nullifier = compute_siloed_nullifier(consumer, nullifier);
         let lower_wit =
-            get_low_nullifier_membership_witness(context.block_number(), siloed_nullifier);
+            get_low_nullifier_membership_witness(self.context.block_number(), siloed_nullifier);
```

Comment:
Didn't expose block_number directly on `self` as it doesn't seem to be very used. It's used only in 2 non-test, non-protocol contracts in our codebase:

<img width="572" height="307" alt="Image" src="https://github.com/user-attachments/assets/8e216a6d-6ac6-4d8a-95cd-220fd9809620" />

Not dying on this hill though.